auth0/resource_data.go

```diff
@@ -92,6 +92,17 @@ func Slice(d Data, key string) (s []interface{}) {
 	return
 }
 
+// Set accesses the value held by key and type asserts it to a set.
+func Set(d Data, key string) (s *schema.Set) {
+	if d.HasChange(key) {
+		v, ok := d.GetOkExists(key)
+		if ok {
+			s = v.(*schema.Set)
+		}
+	}
+	return
+}
+
```

Comment:
Perhaps we could follow the approach used by `List()` and return an iterator instead? Since TypeList and TypeSet are both part of the [aggregate types](https://www.terraform.io/docs/extend/schemas/schema-types.html#aggregate-types) maybe its best for them to behave similarly.

We could create a new method for `&iterator` if you like to return the raw slice so you can achieve a similar result as what you currently have. For example `Set(d, "enabled_clients").Slice()`.

What do you think?

Reply:
We could also move its definition closer to the `iterator` maybe right after `List()`.

Reply:
At first glance it felt a little over the top for the simple case I'm implementing here, but having worked it through I can see that it's nice to make things consistent at least, and I expect the `Slice()` function may make the iterator returned from `List()` a bit easier to use as well.  I implemented the simple fix in a first commit and your suggested refactoring in a second commit, then finally moved it down in a third, to keep the diffs simple and in case you actually prefer the simpler solution, in which case we can revert the second two commits.